Refactor : mainList 데이터 요청 응답 코드 리팩토링 및 server , client 컴포넌트 분리 #16

dv-yeop920 opened the pull request:
- [x] mainList 데이터 요청 응답 코드 리팩토링
- [x] server 와 client 컴포넌트 분리

refactor/list -> develop

서버 컴포넌트를 좀 더 깨끗하게 하기 위해 클라이언트 컴포넌트를 따로 만들었습니다.
api에 요청 하는 로직은 함수로 따로 분류 하였고 함수에서 리턴해 주는 값을 변수에 할당 해서 클라이언트 컴포넌트에 props로 데이터를 전달했습니다
close #16

dv-yeop920 committed:
Refactor : mainList 데이터 요청 응답 코드 리팩토링 및 server , client 컴포넌트 분리

- [x] mainList 데이터 요청 응답 코드 리팩토링
- [x] server 와 client 컴포넌트 분리

refactor/list -> develop

서버 컴포넌트를 좀 더 깨끗하게 하기 위해 클라이언트 컴포넌트를 따로 만들었습니다.
api에 요청 하는 로직은 함수로 따로 분류 하였고 함수에서 리턴해 주는 값을 변수에 할당 해서 클라이언트 컴포넌트에 props로 데이터를 전달했습니다

zxxng commented on src/components/mainList/VideoList.tsx:
```diff
@@ -0,0 +1,55 @@
+                  pathname: `/detail/${VIDEO.channelId}`,
+                  query: {
+                    videoInfo: encodeURIComponent(JSON.stringify(VIDEO)),
+                  },
```
쿼리값 수정해야 합니다!
`query: { id: video.id },`

zxxng commented on src/components/mainList/VideoList.tsx:
```diff
@@ -0,0 +1,55 @@
+        {videoList.map((video: IVideo) => {
+          const VIDEO = video.snippet
+          return (
+            <li className={styles.videoCard}>
```
`key={index}` 추가 부탁드립니당

wjdxor133 replied:
해당 코드에서 <></> 를 사용한 이유가 있을까요??
특별한 이유가 없다면, 지워도 될거 같아 여쭤봅니다.

zxxng commented on src/app/mainList/page.tsx:
```diff
@@ -1,56 +1,28 @@
+import VideoList from '@/components/mainList/VideoList'
 
-type VideoList = IVideo[]
+type VideoListType = IVideo[]
+
+const getVideoList = async (): Promise<VideoListType> => {
```
` VideoListType` 선언할 필요 없이
```js
const getVideoList = async (): Promise<IVideo[]> => 
```
이렇게 사용해도 괜찮지 않을까요?!
`IVideo`에서 추가되거나 삭제되는 부분 없이 배열로 사용되는 부분이라 굳이 타입을 새로 지정해주지 않아도 될 것 같습니다!

wjdxor133 replied:
네, @zxxng 님 말씀대로 중복 정의가 코드를 혼란을 일으킬 수 있어,
선언한 타입을 추가로 선언하지 않아도 될 것 같습니다.

zxxng commented on src/app/mainList/page.tsx:
```diff
@@ -1,56 +1,28 @@
-  // const response = await (await axios.get(URL)).data
-  // const videoList: VideoList = response.items
-  const videoList: VideoList = VIDEO_LIST.items
+const VideoListPage: NextPage<ISnippet> = async () => {
+  const videoList: VideoListType = await getVideoList()
+  //const videoList: VideoListType = VIDEO_LIST.items
```
여기도 `VideoListType` 을 `IVideo[]`으로..!

dv-yeop920 committed:
Refactor : li의 키값 및 Link 컴포넌트 query 수정

### Refactor
- [x] li 태그 키값 추가
- [x] Link 컴포넌트 쿼리 수정

### 반영 브랜치
Refactor/list -> develop

### 변경 사항
 리뷰 해주신 li의 키값 및 Link 컴포넌트 query 수정 했습니다
